### PR TITLE
Fix instructions on how to run the inference server variant 1 

### PR DESCRIPTION
**Pull request by Rudd-O**
Fixes https://github.com/LAION-AI/Open-Assistant/issues/2594 .

### CODE REVIEW

**Comment by Rudd-O** [comment]
I haven't yet figured out how to make the models downloaded by the worker persist across dockercompose up and dockercompose down.  If a kind soul would let me know what needs to be done in order for the user to not have to redownload the models every time the stack starts, I would be so happy and I would add it to the README too.

**Comment by AbdBarho** [comment]
probably like this:

```yml
# docker-compose.override.yml
services:
  inference-worker:
    volumes:
      - "./oasst-shared:/opt/inference/lib/oasst-shared"
      - "./inference/worker:/opt/inference/worker"
      - worker-cache:/data

volumes:
  worker-cache:
```

I would just like to state again that this should not be necessary unless you wan't develop the website.

**Comment by yk** [comment]
thank you for this! yes I'd add a description that this is strictly a development setup to develop the system and website. quite a number of people are looking to run OA locally.

File: inference/README.md [hunk 1]
```diff
@@ -26,31 +26,37 @@ docker compose --profile inference up -d
...
-> the name of the `text-client` container started in step 2.
-
 > **Note:** The compose file contains the bind mounts enabling you to develop on
 > the modules of the inference stack, and the `oasst-shared` package, without
 > rebuilding.
 
+> **Note:** You can change the model by editing variable `MODEL_CONFIG_NAME` in
+> the `docker-compose.yaml` file. Valid model names can be found at
+> https://huggingface.co/OpenAssistant .
+
```
Comment (farmisen) [inline inference/README.md:43]: huggingface only gives the model names, the valid model config names mapping to the model names, can be found in https://github.com/LAION-AI/Open-Assistant/blob/main/oasst-shared/oasst_shared/model_configs.py#L23

**Comment by pevogam** [comment]
> probably like this:
> 
> ```yaml
> # docker-compose.override.yml
> services:
>   inference-worker:
>     volumes:
>       - "./oasst-shared:/opt/inference/lib/oasst-shared"
>       - "./inference/worker:/opt/inference/worker"
>       - worker-cache:/data
> 
> volumes:
>   worker-cache:
> ```
> 
> I would just like to state again that this should not be necessary unless you wan't develop the website.

@Rudd-O Are you planning to integrate this into the current PR or should we all rely on overriding YAML config to achieve it?

File: inference/README.md (hunk 1 above)
Comment (Rudd-O) [reply inference/README.md:43]: Will fix.

**Comment by Rudd-O** [comment]
@AbdBarho 

> probably like this:

No, I don't think that works.  Stuff is stored in a bunch of places including `~/cache` according to the `worker_full_main.sh` script.  It isn't clear to me where.

This stuff needs to be cleaned up for real, as in formally define where these programs are storing downloaded data — so that users can define mounts that will preserve data across runs.

**Comment by Rudd-O** [comment]
> @Rudd-O Are you planning to integrate this into the current PR or should we all rely on overriding YAML config to achieve it?

Not planning to integrate these, the instructions seem false to me.

File: inference/README.md [hunk 2]
```diff
@@ -26,31 +29,37 @@ docker compose --profile inference up -d
 Tail the logs:
 
 ```shell
+# cd /root/of/this/repository
 docker compose logs -f    \
     inference-server      \
     inference-worker      \
-    inference-text-client \
-    inference-text-generation-server
+    inference-safety
```
Comment (olliestanley) [inline inference/README.md:36]: inference-safety is removed from inference profile in recent main

Comment (Rudd-O) [reply inference/README.md:36]: OK.  I will kill it here too.

**Comment by pevogam** [comment]
> @AbdBarho
> 
> > probably like this:
> 
> No, I don't think that works. Stuff is stored in a bunch of places including `~/.cache` according to the `worker_full_main.sh` script. It isn't clear to me where.
> 
> This stuff needs to be cleaned up for real, as in formally define where these programs are storing downloaded data — so that users can define mounts that will preserve data across runs.

Hmmm, it worked just fine for me with
```diff
diff --git a/docker-compose.yaml b/docker-compose.yaml
index c6333dac..345df8f4 100644
--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -252,6 +252,7 @@ services:
     volumes:
       - "./oasst-shared:/opt/inference/lib/oasst-shared"
       - "./inference/worker:/opt/inference/worker"
+      - worker-cache:/data
     deploy:
       replicas: 1
     profiles: ["inference"]
@@ -335,3 +336,4 @@ volumes:
   netdataconfig:
   netdatalib:
   netdatacache:
+  worker-cache:
```
So if you want you can try to see if the model is always downloaded for you with a diff like this one.

**Comment by olliestanley** [comment]
These changes were integrated into #2969, which I have added a co-author tag to for @Rudd-O. Thanks :)

**Pull request closed without merging**